### PR TITLE
Fix the `failOnError not found` error when building Spark 3.4.0

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi <13716567376yh@gmail.com>
Related to #6383 

## Rationale of the PR
In Spark3.4.0, `GetMapValue` doesn't have the field `failOnError` and always return null on invalid access. In other words, `failOnError` is always `false` in Spark 3.4.0. This patch fix this compile error when building Spark 3.4.0:
### branch 22.12
```
...
[ERROR] [Error] /home/remziy/working/rapids/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala:2596: value failOnError is not a member of org.apache.spark.sql.catalyst.expressions.GetMapValue
[ERROR] 24 errors found
```

### this patch
```
...
[ERROR] 23 errors found
```

## Changes in this PR
Define a new abstract class `GetMapValueMeta` in the shim layer of `spark311~340` and `spark340+`, in which we redefine the `convertToGpu` function. For Spark 3.4.0, we always set `GpuGetMapVale.failOnError` as `false`.

## Attention
This patch doesn't update `GpuGetMapValue`.

### CODE REVIEW

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala [hunk 1]
```diff
@@ -2585,16 +2585,15 @@ object GpuOverrides extends Logging {
...
           if (isLit(in.left) && (!isLit(in.right))) {
-            willNotWorkOnGpu("Looking up Map Scalars with Key Vectors " +
+          willNotWorkOnGpu("Looking up Map Scalars with Key Vectors " +
               "is not currently unsupported.")
-          }
         }
-        override def convertToGpu(map: Expression, key: Expression): GpuExpression =
-          GpuGetMapValue(map, key, in.failOnError)
-      }),
+    }
```
Comment (tgravescs) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala:2594]: spacing looks off here as well

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala [hunk 2]
```diff
@@ -2585,16 +2585,15 @@ object GpuOverrides extends Logging {
           TypeSig.NULL + TypeSig.DECIMAL_128 + TypeSig.MAP + TypeSig.BINARY),
           TypeSig.MAP.nested(TypeSig.all)),
         ("key", TypeSig.commonCudfTypes + TypeSig.DECIMAL_128, TypeSig.all)),
-      (in, conf, p, r) => new BinaryExprMeta[GetMapValue](in, conf, p, r) {
+      (in, conf, p, r) => new GetMapValueMeta(in, conf, p, r) {
         override def tagExprForGpu(): Unit = {
           if (isLit(in.left) && (!isLit(in.right))) {
-            willNotWorkOnGpu("Looking up Map Scalars with Key Vectors " +
+          willNotWorkOnGpu("Looking up Map Scalars with Key Vectors " +
```
[the remark this answers is not in the document]
Comment (HaoYang670) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala]: updated

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala (hunk 1 above)
Comment (HaoYang670) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala]: updated

**Comment by HaoYang670** [comment]
build

File: sql-plugin/src/main/311until340-all/scala/com/nvidia/spark/rapids/shims/GetMapValueMeta.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ */
+abstract class GetMapValueMeta (
+    expr: GetMapValue,
+    conf: RapidsConf,
+    parent: Option[RapidsMeta[_, _, _]],
+    rule: DataFromReplacementRule)
+  extends BinaryExprMeta[GetMapValue](expr, conf, parent, rule) {
+
+    override def convertToGpu(map: Expression, key: Expression): GpuExpression =
+        GpuGetMapValue(map, key, expr.failOnError)
```
Comment (tgravescs) [inline sql-plugin/src/main/311until340-all/scala/com/nvidia/spark/rapids/shims/GetMapValueMeta.scala:37]: nit sorry missed this but this should be 2 space indentation.   Same with the other file.  Not required to change if you don't want but fyi for future

**Review by tgravescs — APPROVED** [review]

**Pull request merged**